src/main/antlr4/Expr.g4

```diff
@@ -0,0 +1,65 @@
+grammar Expr;
+
+@header {
+package io.mfj.expr.antlr4;
+}
+
+root : expression EOF;
+
+expression: term ( CONJUNCTION expression )?;
+
+term : statement | parens | not;
+
+parens: '(' expression ')';
+
+not : notStart expression ')';
+notStart : NOTP | notSpaceP;
+fragment NOTP : [nN][oO][tT]'(';
+notSpaceP : NOT '(';
+NOT: [nN][oO][tT];
+
+statement : value OPERATOR value ( OPERATOR value )?;
+
+value : value MATH_OP value | varName | literalValue | list;
+
+CONJUNCTION: AND | OR;
+fragment AND : [aA][nN][dD];
+fragment OR : [oO][rR];
+
+OPERATOR : '!=' | '<>' | '>=' | '<=' | '>' | '<' | '=~' | '=' | IN | CONTAINS ;
+
+fragment IN : '!'?[iI][nN];
+fragment CONTAINS: '!'?[cC][oO][nN][tT][aA][iI][nN][sS];
+
+literalValue : nul | number | string | bool | regex | date | time | datetime;
+
+nul: 'null';
+
+string : '"' (ESC|.)*? '"';
+fragment ESC : '\\"' | '\\\\';
+
+regex : '/' (RESC|.)*? '/';
+fragment RESC : '\\/' | '\\\\';
+
+number: INT | DECIMAL;
+INT : '-'? DIGIT+;
+DECIMAL : '-'? DIGIT* '.' DIGIT*;
+DIGIT: [0-9];
+
+bool : TRUE | FALSE;
+TRUE: 'true';
+FALSE: 'false';
+
+date : 'd\'' (SQESC|':'|.)*? '\'';
+time : 't\'' (SQESC|':'|.)*? '\'';
+datetime: 'dt\'' (SQESC|':'|.)*? '\'';
+fragment SQESC: '\\\'' | '\\\\';
+
+varName: VAR_NAME;
+VAR_NAME: [a-zA-Z_][a-zA-Z0-9_]*;
+
+list : '[' ( value ( ',' value )+ ','? )? ']';
```

Comment:
This is not identical to the expression in the readme, there's a `+` v `*`. Got me thinking about some more tests. I added them, take them or leave them as you like. They passed, for what it's worth.

https://github.com/measuresforjustice/expr/pull/11

Reply:
The last `?` makes it work. I merged in your tests at 5077350. They are definitely good things to test, thank you.